internal/exec/util/blkid.c

```diff
@@ -248,3 +254,39 @@ static result_t extract_part_info(blkid_partition part, struct partition_info *i
 
 	return RESULT_OK;
 }
+
+// blkid_get_block_devices fetches block devices with the given FSTYPE
+result_t blkid_get_block_devices(const char *fstype, struct block_device_list *device) {
+	blkid_dev_iterate iter _cleanup_iterator_ = NULL;
+	blkid_dev dev;
+	blkid_cache cache _cleanup_cache_ = NULL;
+	int err, count = 0;
+	const char *ctmp = NULL;
+	
+	if (blkid_get_cache(&cache, "/dev/null") != 0)
+	   return RESULT_GET_CACHE_FAILED;
+	
+	if (blkid_probe_all(cache) != 0)
+       return RESULT_PROBE_FAILED;
+
+	iter = blkid_dev_iterate_begin(cache);
+	
+	blkid_dev_set_search(iter, "TYPE", fstype);
+	
+	while (blkid_dev_next(iter, &dev) == 0) {
+	    dev = blkid_verify(cache, dev);
+	    if (!dev)
+		continue;
+	    ctmp = blkid_dev_devname(dev);
+	    if (count > MAX_BLOCK_DEVICES)
```

Comment:
Off-by-one error: `device->path[MAX_BLOCK_DEVICES]` is one past the end of the array.